### ISSUE TITLE
Artifactory NuGet Restore not including local package cache

### ISSUE THREAD

**Issue by Skinner007** [jfrog/artifactory-azure-devops-extension#160]
When using the Artifactory NuGet task with the restore command, the local cache is not being included as a source. This results in every package being downloaded on every execution, which is incredibly inefficient.

The logs for the Artifactory NuGet task show (values masked for privacy):

NuGet Config files used:
C:\Users\\*\AppData\Local\Temp\jfrog.cli.751553911\jfrog.cli.nuget.343174762

Feeds used:
https://\*/api/nuget/\*_nuget 

The standard NuGet task does properly include the local cache, and logs show (values masked for privacy):

NuGet Config files used:
D:\\*\Agent1\_work\16\Nuget\tempNuGet_8519.config

Feeds used:
**C:\Users\\*\\.nuget\packages\\**
https://*/nuget/v3/index.json

The logs clearly show the local package cache is not being included by the Artifactory NuGet task. The checkbox on the task to "Disable local NuGet cache" appears to have no impact on the local cache being excluded.

TFS On-Prem 2018
JFrom Extension version 1.6.2

**Comment by liron-shalom** [comment]
@Skinner007, thanks for reporting this issue.
Can you please provide us the difference between the two configurations?

**Comment by Skinner007** [comment]
I'm not entirely sure what configuration difference you're looking for. The OOTB nuget task is typically configured to point to a solution file and a nuget.config or to a TFS package feed for source repository mapping. When executed, it includes the path to the local cache as the first source so packages are pulled from there before going to the other package feeds.

The Artifactory task is configured to point to an Artifactory repository as the source. It too is point to a solution file and provided the Destination Directory. For .net core solutions, you can set the Destination directory to the local cache and everything works fine due to the changes in how .net core references packages. For .net framework solutions, this approach does not work. You must set the destination to "Packages" for the packages folder to be created, which then results in all packages being downloaded since the cache is not listed as the primary feed.

In short, the package cache folder needs to be added as the primary feed in the config file this task creates, so packages will be pulled from there first for .net framework solutions.

**Comment by liron-shalom** [comment]
@Skinner007 thanks for your comment.
We found a bug which cause "Disable local NuGet cache" to have no impact on the local cache being excluded.
Can you please try the workaround described in [this issue](https://github.com/jfrog/artifactory-azure-devops-extension/issues/164) and tell us if its resolve the issue?

**Comment by Skinner007** [comment]
That work around would only work with yaml definitions, not classic. We're using classic, so the bug would need to be fixed to be resolved
